python/cudf_polars/cudf_polars/containers/dataframe.py

```diff
@@ -83,6 +93,35 @@ def from_cudf(cls, df: cudf.DataFrame) -> Self:
             ]
         )
 
+    @classmethod
+    def from_polars(cls, df: pl.DataFrame) -> Self:
+        """
+        Create from a polars dataframe.
+
+        Parameters
+        ----------
+        df
+            Polars dataframe to convert
+
+        Returns
+        -------
+        New dataframe representing the input.
+        """
+        table = df.to_arrow()
+        schema = table.schema
+        for i, field in enumerate(schema):
+            schema = schema.set(
+                i, pa.field(field.name, dtypes.downcast_arrow_lists(field.type))
+            )
+        # No-op if the schema is unchanged.
+        d_table = plc.interop.from_arrow(table.cast(schema))
+        return cls(
+            [
+                NamedColumn(column, h_col.name).copy_metadata(h_col)
```

Comment:
could the name be considered metadata at this point?

Reply:
A `NamedColumn` always needs a name, but doesn't always need metadata.

Reply:
I suppose a more focused question would have been should `copy_metadata` also copy the name if it exists?

Reply:
Then we run into Liskov issues because `copy_metadata` is a method on `Column` not `NamedColumn`, and the former doesn't have a name.

I want to fix this elsewhere, but that needs more thought.